rust/connlib/shared/src/lib.rs

```diff
@@ -9,6 +9,12 @@ pub mod control;
 pub mod error;
 pub mod messages;
 
+/// Module to generate and store a persistent device ID on disk
+///
+/// Only properly implemented on Linux and Windows (platforms with Tauri and headless client)
+pub mod device_id;
+
+// Must be compiled on Mac so the Mac runner can do `version-check` on `linux-client`
```

Comment:
Hmm I can refactor that if it continues to be a nuissance.

Reply:
I don't remember what I needed it for, but I did just stub out the Tauri client for macOS in #3977 so I can do some of the checks locally without waiting on CI. So that might go in soon.

Reply:
We could rename `linux-client` to `cli-client` too so it's not weird to say that a Mac might compile the "Linux" client even if headless Mac clients are not supported for production.